lib/core/src/Cardano/SharedWallet/DB/Sqlite/TH.hs

```diff
@@ -0,0 +1,86 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+-- |
+-- Copyright: © 2018-2021 IOHK
+-- License: Apache-2.0
+--
+-- Auto-generated Sqlite & Persistent machinery via Template-Haskell. This has
+-- been moved into a separate file so that we can treat it slightly differently
+-- when computing code-coverage.
+
+module Cardano.SharedWallet.DB.Sqlite.TH where
+
+import Prelude
+
+import Cardano.Address.Script
+    ( Cosigner, Script )
+import Cardano.SharedWallet.Script
+    ( CredentialType )
+import Cardano.SharedWallet.SharedState
+    ( SharedWalletState )
+import Cardano.Wallet.DB.Sqlite.Types
+    ( BlockId, sqlSettings' )
+import Data.Text
+    ( Text )
+import Data.Time.Clock
+    ( UTCTime )
+import Data.Word
+    ( Word32, Word8 )
+import Database.Persist.Class
+    ( AtLeastOneUniqueKey (..), OnlyOneUniqueKey (..) )
+import Database.Persist.TH
+    ( mkDeleteCascade, mkMigrate, mkPersist, persistLowerCase, share )
+import GHC.Generics
+    ( Generic (..) )
+
+import qualified Cardano.Wallet.Primitive.AddressDiscovery.Sequential as W
+import qualified Cardano.Wallet.Primitive.Types as W
+import qualified Data.ByteString.Char8 as B8
+
+share
+    [ mkPersist sqlSettings'
+    , mkDeleteCascade sqlSettings'
+    , mkMigrate "migrateAll"
+    ]
+    [persistLowerCase|
+
+-- Shared Wallet
+SharedWallet
```

Comment:
Is there a good case for making an entirely separate database schema?
It seems to me that shared wallet data could just be another table.

Reply:
yes, shared wallet could be separate, light entity, that could follow shared addresses, its balance, will be able to sign txs using the functionality we are going to support soon (ie., standalone signing). Having this imperative in mind (@KtorZ please confirm if I got this right) it was tempting to separate shared wallet as much as possible from beginning (to avoid future  heavy refactorings and decouplings)

Reply:
reconsidered here after discussion with @KtorZ I will move DB stuff under Wallet hood - as we intend to reuse a lot from sequential wallet